File: 0002-add-two-numbers/0002-add-two-numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode dummy = new ListNode(0);
        ListNode temp = dummy;
        int carry =0;
        while(l1 != null || l2  != null || carry != 0){
            int sm =0;
            if(l1 != null){
                sm += l1.val;
                l1 = l1.next;
            }
            if(l2 != null){
                sm += l2.val;
                l2 = l2.next;
            }
            sm += carry;
            carry = sm / 10;
            ListNode temporary = new ListNode(sm%10);
            temp.next = temporary;
            temp = temp.next;
        }
        return dummy.next;
    }
}

File: 0002-add-two-numbers/0002-add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        carry = 0
        dummy = temp = ListNode()
        while(l1 or l2 or carry):
            sm  = 0
            if l1:
                sm += l1.val
                l1 = l1.next
            if l2:
                sm += l2.val
                l2 = l2.next
            sm += carry
            carry = sm // 10
            node = ListNode(sm % 10)
            temp.next = node
            temp = temp.next
        return dummy.next
